new/cookbook-segment-deprecation.md

```diff
@@ -0,0 +1,55 @@
+---
+RFC: unassigned
+Author: Lamont Granquist <lamont@chef.io>
+Status: Draft
+Type: Standards Track
+---
+
+# Title
+
+Cookbook Segment Deprecation
+
+## Motivation
+
+    As a Chef User/Developer
+    I want to be able to extend the structure of cookbooks easily,
+    in order to quickly adapt to new automation needs.
+
+## Specification
+
+This RFC will add a special `all_files` segment to cookbooks which will duplicate (initially)
+all of the existing cookbook manifest along with arbitrary files outside of the current scope of
+any of the cookbook segments.  The Chef Server will need to be patched in order to support this
+new 'segment' and accept it since the Chef Server currently does strict enforcement of file
+segments server-side.  This will also generate a version bump of the Chef Server protocol which
+will be used by servers to determine if the Chef Client supports this field or not.  Clients MAY
+implement code which determines the acceptable API version of the Server and then only send
+the new API requests to the server.  Clients MAY instead send requests which include both the
+old segment and new segment data.  The Server MUST detect the Client's desired API version and
```

Comment:
^ that comment is out of date, but i'm still seeing it on my GH page even though I've pushed a complete rewrite.